Integration/AndroidAAR/BabylonNative/src/main/java/BabylonNative/BabylonView.java

```diff
@@ -0,0 +1,113 @@
+package BabylonNative;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class BabylonView extends SurfaceView implements SurfaceHolder.Callback2, View.OnTouchListener {
+    private static final String TAG = "BabylonView";
+    private boolean mViewReady = false;
+    private ViewDelegate mViewDelegate;
+
+    public BabylonView(Context context, ViewDelegate viewDelegate) {
+        super(context);
+        init(viewDelegate);
+        BabylonNative.Wrapper.initEngine(context.getResources().getAssets());
+    }
+
+    private void init(ViewDelegate viewDelegate) {
+        SurfaceHolder holder = getHolder();
+        holder.addCallback(this);
+        setOnTouchListener(this);
+        mViewDelegate = viewDelegate;
+    }
+
+    public void loadScript(String path) {
+        BabylonNative.Wrapper.loadScript(path);
+    }
+
+    public void eval(String source, String sourceURL) {
+        BabylonNative.Wrapper.eval(source, sourceURL);
+    }
+
+    public void onPause() {
+        setVisibility(View.GONE);
+        BabylonNative.Wrapper.activityOnPause();
+    }
+
+    public void onResume() {
```

Comment:
Do we need a corresponding setVisibility call here?

Reply:
It's done in the activity based on another method (onWindowFocusChanged). I will check when working on views/surface if it can be simplified.